extensions/emmet/src/abbreviationActions.ts

```diff
@@ -18,6 +19,16 @@ const hexColorRegex = /^#[\da-fA-F]{0,6}$/;
 // 	's', 'samp', 'select', 'small', 'span', 'strike', 'strong', 'sub', 'sup',
 // 	'textarea', 'tt', 'u', 'var'];
 
+
+/**
+ * Stores previous expansion attempts for Wrap with Abbreviation commands.
+ * Both successful (accepted) and non-successful (cancelled) expansion attempts are stored.
+ * The expansion attempts are stored per extension reload.
+ */
+let previousItems: Set<string> = new Set();
```

Comment:
could be const if you don't recreate

Reply:
should this be initialized from previous sessions. e.g. if I press up in the commit message box, i can see commit messages from long ago

Reply:
I think that can go into a separate feature, since firstly, I'd have to look into where to write the abbreviation history to, and secondly, I'd probably need to add a "clear history" command.